src/librustc/middle/ty.rs

```diff
@@ -282,6 +282,7 @@ pub enum Variance {
 #[derive(Clone, Debug)]
 pub enum AutoAdjustment<'tcx> {
     AdjustReifyFnPointer(ast::DefId), // go from a fn-item type to a fn-pointer type
+    AdjustUnsafeFnPointer, // go from a safe fn pointer to an unsafe fn pointer
```

Comment:
Whoa, you can match against this with `ty::AdjustUnsafeFnPointer(..)`, that's an interesting... feature.